### PR TITLE
Add hackathon page

### PR DESCRIPTION
**Pull request by razvand**
Add general information about hackathon. Add challenge description and scoring.

### CODE REVIEW

File: content/en/docs/sessions/hackathon.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+**Id**: `lib-bjoern`
+
+**Description**: Port [Bjoern](https://github.com/jonashaag/bjoern) as a Unikraft library.
+
+**Points**: 20 points
+
+**Links / Resources**:
+
+* [Bjoern](https://github.com/php/php-src)
```
Comment (gabrielmocanu) [inline content/en/docs/sessions/hackathon.md:192]:
```suggestion
* [Bjoern](https://github.com/jonashaag/bjoern)
```

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: content/en/docs/hackathon.md [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+
+Additional points will be awarded for:
+
+ * Signing off your commits with `git commit -s`, as part of [Developer
+    Certificate of
+    Origin](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin) (1
+    point per commit) (note, teams of multiple people should have multiple sign-offs);
+ * Every commit should include a precise explanation of the changes made (1
+   point per commit);
+ * Every commit of the solution should be leave the project in a working state
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:34]:
```suggestion
 * Every commit of the solution should leave the project in a working state
```

File: content/en/docs/hackathon.md [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+the Unikraft core repository.  New tests should be introduced as new files under
+a new directory `tests/` of internal microlibraries.  For example, for
+`vfscore`, a new directory will be located at `libs/vfscore/tests/`.
+
+To create a suite, for, for example, `stat()` in `vfscore`, introduce a new file
+`test_stat.c` where you register the suite, cases and create expectations from
+the `stat()` syscall provided by `vfscore`.  Expectations and tests can usually
+be checked by programming different scenarios and checking whether the return
+code or `errno` is set correctly.  Check out the relevant POSIX document for
+function in question.
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:58]:
```suggestion
the function in question.
```

File: content/en/docs/hackathon.md [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+
+### 3. Add ukstore Entries
+
+| ID | Points | Short description |
+|----|--------|-------------------|
+| `ukstore-entries` | 1 point per entry | Add new [ukstore](https://github.com/unikraft/unikraft/pull/275) entries. |
+
+#### Description
+
+`ukstore` is a new internal library for storing and retreving information, for
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:101]:
```suggestion
`ukstore` is a new internal library for storing and retrieving information, for
```

File: content/en/docs/hackathon.md [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+
+### 3. Add new unit tests to `kraft`
+
+| ID | Points | Short description |
+|----|--------|-------------------|
+| `kraft-unit-tests` | 1 point per test | Add new tests in [kraft](https://github.com/unikraft/kraft). |
+
+#### Description
+
+`kraft` is a Python-based toolchain and acts as the companion tool for manging,
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:122]:
```suggestion
`kraft` is a Python-based toolchain and acts as the companion tool for managing,
```

File: content/en/docs/hackathon.md [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+
+### 13. Port Memcached to Unikraft
+
+| ID | Points | Short description |
+|----|--------|-------------------|
+| `lib-memcached` | 25 points | Port [memcached](https://github.com/memcached/memcached) as a Unikraft library. |
+
+#### Description
+
+[memecached](https://memcached.org/) is a general purpose key-value store.  Port
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:403]:
```suggestion
[memcached](https://memcached.org/) is a general purpose key-value store.  Port
```

File: content/en/docs/hackathon.md [hunk 7]
```diff
@@ -0,0 +1,487 @@
...
+kraft list add https://github.com/$USERNAME/app-memcached.git
+kraft list update
+kraft init -t memcached@staging ./my-memcached-app
+```
+
+
+#### Links and additional resources
+
+* [memcached source code](https://github.com/memcached/memcached)
+* [memecached main website](https://memcached.org/)
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:417]:
```suggestion
* [memcached main website](https://memcached.org/)
```

File: content/en/docs/hackathon.md [hunk 8]
```diff
@@ -0,0 +1,487 @@
...
+
+
+***
+
+
+### 9. Port MySQL to Unikraft
+
+| ID | Points | Short description |
+|----|--------|-------------------|
+| `lib-mysql` | 25 points| Port [MySQl](https://github.com/mysql/mysql-server) as a Unikraft library. |
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:291]:
```suggestion
| `lib-mysql` | 25 points| Port [MySQL](https://github.com/mysql/mysql-server) as a Unikraft library. |
```

File: content/en/docs/hackathon.md [hunk 9]
```diff
@@ -0,0 +1,487 @@
...
+
+```bash
+kraft list add https://github.com/$USERNAME/app-mysql.git
+kraft list update
+kraft init -t mysql@staging ./my-mysql-app
+```
+
+#### Links and additional resources
+
+* [MySQl source code](https://github.com/mysql/mysql-server)
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:308]:
```suggestion
* [MySQL source code](https://github.com/mysql/mysql-server)
```

File: content/en/docs/hackathon.md [hunk 10]
```diff
@@ -0,0 +1,487 @@
...
+* [MySQl main website](https://www.mysql.com/)
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:309]:
```suggestion
* [MySQL main website](https://www.mysql.com/)
```

File: content/en/docs/hackathon.md [hunk 11]
```diff
@@ -0,0 +1,487 @@
...
+
+```bash
+kraft list add https://github.com/$USERNAME/app-bjoern.git
+kraft list update
+kraft init -t bjoern@staging ./my-bjoern-app
+```
+
+#### Links and additional resources
+
+* [Bjoern source code](https://github.com/php/php-src)
```
Comment (florin-diaconescu) [inline content/en/docs/hackathon.md:363]:
```suggestion
* [Bjoern source code](https://github.com/jonashaag/bjoern)
```

File: content/en/docs/hackathon.md [hunk 12]
```diff
@@ -0,0 +1,457 @@
...
+### 16. Fix an Open Bug in the Core
+
+| ID | Points | Short description |
+|----|--------|-------------------|
+| `bug-fix` | Depends on bug, contact TA. | Fix an internal bug in the core. |
+
+#### Description
+
+There are a number of outstanding issues/bugs with the Unikraft core repository.
+To help increase the stability and solve problems for edge cases and other issues, solve an open issue which has been reported.
```
Comment (gabrielmocanu) [inline content/en/docs/hackathon.md:453]:
```suggestion
To help increase the stability and solve problems for edge cases and other issues, solve an open issue that has been reported.
```

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: content/en/docs/hackathon.md [hunk 13]
```diff
@@ -0,0 +1,457 @@
...
+|----|--------|-------------------|
+| `uktest-entries` | 1 point per `*EXPECT*` | Use [uktest](/docs/sessions/06-testing-unikraft/#03-unikrafts-testing-framework.md) to add new tests in Unikraft components. |
+
+#### Description
+
+In this challenge, introduce new test suites, cases and assertions/expects in the Unikraft core repository.
+New tests should be introduced as new files under a new directory `tests/` of internal microlibraries.
+For example, for `vfscore`, a new directory will be located at `libs/vfscore/tests/`.
+
+To create a suite, for, for example, `stat()` in `vfscore`, introduce a new file `test_stat.c` where you register the suite, cases and create expectations from the `stat()` syscall provided by `vfscore`.
```
Comment (mogasergiu) [inline content/en/docs/hackathon.md:45]:
```suggestion
To create a suite, for, to give an example, `stat()` in `vfscore`, introduce a new file `test_stat.c` where you register the suite, cases and create expectations from the `stat()` syscall provided by `vfscore`.
```

File: content/en/docs/hackathon.md [hunk 14]
```diff
@@ -0,0 +1,457 @@
...
+
+### 3. Add ukstore Entries
+
+| ID | Points | Short description |
+|----|--------|-------------------|
+| `ukstore-entries` | 1 point per entry | Add new [ukstore](https://github.com/unikraft/unikraft/pull/275) entries. |
+
+#### Description
+
+`ukstore` is a new internal library for storing and retrieving information, for example, statistics or state.
```
Comment (mogasergiu) [inline content/en/docs/hackathon.md:88]:
```suggestion
`ukstore` is a new internal library for storing and retrieving information, such as statistics or state.
```

File: content/en/docs/hackathon.md [hunk 15]
```diff
@@ -0,0 +1,457 @@
...
+
+#### Description
+
+`ukstore` is a new internal library for storing and retrieving information, such as statistics or state.
+Introduce new counters, stats, or states via `ukstore` in other libraries that generate information.
+
+#### Links and Additional Resources
+
+* [ukstore](https://github.com/unikraft/unikraft/pull/275) Library
+* [Example of adding new counters to `uknetdev`](#)
```
Comment (nderjung) [inline content/en/docs/hackathon.md:94]: TODO: Add correct link

cc. @craciunoiuc

**Pull request merged**